Source/WTF/wtf/text/CStringView.h

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2018 Yusuke Suzuki <utatane.tea@gmail.com>
+ * Copyright (C) 2024 Apple Inc. All Rights Reserved.
+ * Copyright (C) 2025 Comcast Inc.
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <span>
+#include <string>
+#include <type_traits>
+#include <wtf/Compiler.h>
+#include <wtf/ForbidHeapAllocation.h>
+#include <wtf/Forward.h>
+#include <wtf/HashFunctions.h>
+#include <wtf/StdLibExtras.h>
+#include <wtf/text/ASCIILiteral.h>
+#include <wtf/text/SuperFastHash.h>
+
+namespace WTF {
+
+class PrintStream;
+class String;
+
+class CStringView final {
+    WTF_FORBID_HEAP_ALLOCATION;
+public:
+    static CStringView fromLatin1(const char* string LIFETIME_BOUND)
+    {
+        if (!string)
+            return CStringView();
+        return CStringView(unsafeMakeSpan(static_cast<const LChar*>(static_cast<const void*>(string)), std::char_traits<char>::length(string) + 1));
```

Comment:
The intermediate cast to `void*` here is surprising. What's the purpose of that?

Reply:
If I recall correctly, at least GCC can't convert char to LChar. I'll try again without it though in case I missed something.

Reply:
Nope, I can't. Compiler complains. StringView transforms into void* to I did that transformation first.

Reply:
`byteCast` did the job.